flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/source/utils/BsonUtilsTest.java

```diff
@@ -0,0 +1,225 @@
+import org.bson.types.Decimal128;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+
+/** Unit test cases for <code>BsonUtils</code> utility class. */
+public class BsonUtilsTest {
+
+    @Test
+    public void testCompareBsonValue() {
+        // test compare Decimal128
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonDecimal128(Decimal128.parse("18")),
+                                new BsonDecimal128(Decimal128.parse("17")))
+                        > 0);
+        assertEquals(
+                0,
+                BsonUtils.compareBsonValue(
+                        new BsonDecimal128(Decimal128.parse("17")),
+                        new BsonDecimal128(Decimal128.parse("17"))));
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonDecimal128(Decimal128.parse("16")),
+                                new BsonDecimal128(Decimal128.parse("17")))
+                        < 0);
+
+        // test compare String
+        assertTrue(
+                BsonUtils.compareBsonValue(new BsonString("apple"), new BsonString("banana")) < 0);
+
+        assertEquals(
+                0, BsonUtils.compareBsonValue(new BsonString("banana"), new BsonString("banana")));
+
+        assertTrue(
+                BsonUtils.compareBsonValue(new BsonString("cherry"), new BsonString("banana")) > 0);
+
+        // test compare Array
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonArray(
+                                        Arrays.asList(
+                                                new BsonString("fruit"), new BsonString("apple"))),
+                                new BsonArray(
+                                        Arrays.asList(
+                                                new BsonString("fruit"), new BsonString("banana"))))
+                        < 0);
+
+        assertEquals(
+                0,
+                BsonUtils.compareBsonValue(
+                        new BsonArray(
+                                Arrays.asList(new BsonString("fruit"), new BsonString("banana"))),
+                        new BsonArray(
+                                Arrays.asList(new BsonString("fruit"), new BsonString("banana")))));
+
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonArray(
+                                        Arrays.asList(
+                                                new BsonString("fruit"), new BsonString("cherry"))),
+                                new BsonArray(
+                                        Arrays.asList(
+                                                new BsonString("fruit"), new BsonString("banana"))))
+                        > 0);
+
+        // test compare Binary
+        assertTrue(
```

Comment:
1. what happens if same array with different order?
2. It seems that arrays compare just min value, it's ok? See BsonUtils#compareBsonArray & BsonUtils#compareBsonValue with isTopLevel.

Reply:
1. They will be sorted before comparison.
2. It's MongoDB's documented comparison rules.

https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/

Added more testcases & comments to cover these cases.